### PR TITLE
Add metrics for vectorize call

### PR DESCRIPTION
**Pull request by Hazel-Datastax**
**What this PR does**:
Add vectorize metrics :
1. number of calls
2. String size in bytes
3. timer/histogram for calls to NeMo or other third-party providers

**Which issue(s) this PR fixes**:
Fixes [Jira C2-3308](https://datastax.jira.com/browse/C2-3308)

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by Hazel-Datastax** [comment]
New vectorize metrics include `vectorize_input_bytes_count` and `vectorize_external_call_duration`, with the `command`, `tenant`, and `embedding_provider` serving as tags. Examples below clarify how these metrics work:

1. Execute `insertMany` twice with the payload below, which includes three $vectorize each time:
```
{
  "insertMany": {
     "documents": [
        { "name": "Coded Cleats", "$vectorize": "ChatGPT integrated sneakers that talk to you"},
        { "name": "Logic Layers", "$vectorize": "An AI quilt to help you sleep forever"},
        { "name": "Vision Vector Frame", "$vectorize": "Vision Vector Frame', 'A deep learning display that controls your mood"}
     ]
   }
}
```
The metrics generated would be as follows:
```
vectorize_external_call_duration_seconds_max{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 0.004752292
vectorize_external_call_duration_seconds_count{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 2.0
vectorize_external_call_duration_seconds_sum{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 0.004887459
// ignore timer histogram

vectorize_input_bytes_max{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 70.0
vectorize_input_bytes_count{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 6.0
vectorize_input_bytes_sum{command="InsertManyCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 302.0
```
Metric Interpretation:

- `vectorize_external_call_duration_seconds_count` is 2 because, despite there being 3 $vectorize directives, they are compiled into a single list for one external vectorize call per insertMany command.
- `vectorize_input_bytes_count` totals 6, reflecting the individual $vectorize instances across two insertMany operations, each containing 3 vectorize.
- `vectorize_input_bytes_sum` aggregates the byte size of the 6 vectorize texts.

2. Execute `findOneAndUpdate` once with the payload below, which includes two $vectorize:
```
{
    "findOneAndUpdate": {
        "sort" : {"$vectorize" : "ChatGPT that talk to you"},
        "update" : {"$set" : {"status" : "active", "$vectorize" : "This is new data"}},
        "options" : {"returnDocument" : "after"}
    }
}
```
The metrics generated would be as follows:
```
vectorize_external_call_duration_seconds_max{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 0.005008792
vectorize_external_call_duration_seconds_count{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 2.0
vectorize_external_call_duration_seconds_sum{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 0.005152583
// ignore timer histogram

vectorize_input_bytes_max{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 24.0
vectorize_input_bytes_count{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 2.0
vectorize_input_bytes_sum{command="FindOneAndUpdateCommand",embedding_provider="HuggingFaceEmbeddingClient",module="sgv2-jsonapi",tenant="unknown",} 40.0
```
Metric Interpretation:
- Differently from `insertMany`, the two $vectorize operations in this command are processed separately, leading to two external vectorize calls. This is reflected in the `vectorize_external_call_duration_seconds_count` metric, which is 2.
- `vectorize_input_bytes_count` is also 2, indicating the presence of two distinct $vectorize fields within the command.
- The `vectorize_input_bytes_sum` metric shows the total byte size of the text from these 2 vectorize fields.

File: src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonApiMetricsConfig.java [hunk 1]
```diff
@@ -70,6 +70,18 @@ public interface JsonApiMetricsConfig {
   @WithDefault("command.processor.process")
   String metricsName();
 
+  @NotBlank
+  @WithDefault("vectorize.external.call.duration")
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonApiMetricsConfig.java:74]: Do we need `external`?

File: src/main/java/io/stargate/sgv2/jsonapi/config/OperationsConfig.java [hunk 2]
```diff
@@ -252,6 +252,6 @@ interface ConsistencyConfig {
     }
   }
   /** @return Flag to enable server side vectorization. */
-  @WithDefault("false")
+  @WithDefault("true")
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/config/OperationsConfig.java:255]: This by default has to be false. If you need it true for any testing add test profile like `EnabledVectorizeProfile`

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import io.stargate.sgv2.api.common.config.MetricsConfig;
+import io.stargate.sgv2.jsonapi.api.request.DataApiRequestInfo;
+import io.stargate.sgv2.jsonapi.api.v1.metrics.JsonApiMetricsConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+
+@ApplicationScoped
+public class MeteredEmbeddingProvider implements EmbeddingProvider {
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java:14]: java docs?

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      JsonApiMetricsConfig jsonApiMetricsConfig,
+      DataApiRequestInfo dataApiRequestInfo,
+      MetricsConfig metricsConfig) {
+    this.meterRegistry = meterRegistry;
+    this.jsonApiMetricsConfig = jsonApiMetricsConfig;
+    this.dataApiRequestInfo = dataApiRequestInfo;
+    tenantConfig = metricsConfig.tenantRequestCounter();
+  }
+
+  public MeteredEmbeddingProvider setEmbeddingClient(
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java:36]: This access pattern is incorrect. You are setting embeddingClient on a ApplicationScoped object. The class needs to be @RequestScoped with providers for every fields or create a regular java object.

File: src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonApiMetricsConfig.java (hunk 1 above)
Comment (Hazel-Datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/api/v1/metrics/JsonApiMetricsConfig.java:74]: Remove `external`

File: src/main/java/io/stargate/sgv2/jsonapi/config/OperationsConfig.java (hunk 2 above)
Comment (Hazel-Datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/config/OperationsConfig.java:255]: I set it to true for testing purposes. Forgot to remove it

File: src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java (hunk 4 above)
Comment (Hazel-Datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java:36]: Changed to `@RequestScoped`

Comment (Hazel-Datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java]: Change to a regular java object

**Review by maheshrajamani — APPROVED** [review]
LGTM

**Pull request merged**